packages/rocketchat-livechat/server/api/v1/agent.js

```diff
@@ -43,13 +43,14 @@ RocketChat.API.v1.addRoute('livechat/agent.next/:token', {
 				department: Match.Maybe(String),
 			});
 
-			const visitor = findGuest(this.urlParams.token);
-			if (!visitor) {
+			const { token } = this.urlParams;
+			const room = findOpenRoom(token);
+			if (room) {
 				throw new Meteor.Error('invalid-token');
 			}
 
 			let { department } = this.queryParams;
-			if (!department) {
+			if (!department || department === 'undefined') {
```

Comment:
can `department` can really be the _string_ `undefined`? if so, why not treat client side?

Reply:
Actually the 'undefined' value is coming from our JS SDK. I have requested some changes on JS SDK and it envolves cases like that. Fixing those methods it will probably fix the issue related to `department` value.